scripts/agency_ingestion/run.py

```diff
@@ -0,0 +1,231 @@
+		except Exception:
+			logging.exception(f"Something went wrong when trying to retrieve {self.name} data")
+			raise
+
+
+class ECDCIngestor(AgencyIngestor):
+
+	def __init__(self):
+		super().__init__("ECDC", ECDC_URL)
+		self.soup = None
+		self.site_content = ""
+		self.json_soup = {}
+
+	def ingest_data(self):
+		logging.info("Ingesting ECDC data")
+		self.get_site_content()
+		self.make_soup()
+		for div in TARGET_DIVS:
+			self.get_json(div)
+			self.process_json(div)
+			self.data_to_csv(FIELDS.get(div))
+			self.store_data(f"{AGENCIES['ECDC']}/{TODAY}_{div}.csv")
+			self.store_data(f"ecdc_{div}_latest.csv")
+
+	def get_site_content(self):
+		logging.info("Getting contents of ECDC website")
+		try:
+			self.site_content = requests.get(ECDC_URL).content.decode("utf-8")
+		except Exception:
+			logging.exception("Something went wrong getting HTML from ECDC website")
+			raise
+
+	def make_soup(self):
+		logging.info("Making BeautifulSoup")
+		try:
+			self.soup = BeautifulSoup(self.site_content, "html5lib")
+		except Exception:
+			try:
+				self.soup = BeautifulSoup(self.site_content, "html.parser")
+			except Exception:
+				logging.exception("Something went wrong trying to make BeautifulSoup")
+				raise
+
+	def get_json(self, div: str):
+		logging.info("Getting HTML from ECDC website")
+		html = self.soup.find("div", id=div)
+		if html is None:
+			raise ValueError(f"div[id='{div}'] not found")
+		script = html.find("script")
+		if script is None:
+			raise ValueError("No JSON data found in div")
+		try:
+			self.json_soup = json.loads(script.contents[0])
+		except Exception:
+			logging.exception("Something went wrong getting JSON from <script>")
+			raise
+
+	def process_json(self, div: str) -> list[dict[str, str | int]]:
+		logging.info("Processing JSON data")
+		for group in self.json_soup["x"]["data"]:
+			text = group["text"]
+			text = text if isinstance(text, list) else [text]
+			# parse each line and remove invalid lines
+			self.data.extend(list(filter(None, map(self.parse_line, text, repeat(div)))))
+
+	@classmethod
+	def parse_line(cls, line: str, div: str) -> [dict[str, str | int]]:
+		if match := re.match(REGEXES.get(div), line):
+			if div == ONSET_OCA_DIV_ID:
+				date, count, country = match.groups()
+				return {"date": date, "count": int(count), "country": country}
+			if div == NOTIF_DIV_ID:
+				date, count = match.groups()
+				return {"date": date, "count": int(count)}
+			if div == ONSET_DATE_DIV_ID:
+				date, count, onset_type = match.groups()
+				return {"date": date, "count": int(count), "type": onset_type}
+		return {}
+
+
+@click.command()
+@click.option("--cdc", is_flag=True, show_default=True, default=False, help="Ingest CDC data")
+@click.option("--ecdc", is_flag=True, show_default=True, default=False, help="Ingest ECDC data")
+@click.option("--who", is_flag=True, show_default=True, default=False, help="Ingest WHO data")
+def run(cdc, who):
```

Comment:
missing ecdc

Reply:
Good catch, I have not tested this outside of Docker+pytest yet!